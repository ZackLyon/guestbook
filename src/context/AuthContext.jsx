import React, { createContext, useContext, useState } from 'react';

const AuthContext = createContext();

const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  const login = ({ username, password }) => {
    const didLoginSucceed =
      username === process.env.USERNAME_AUTH &&
      password === process.env.PASSWORD_AUTH;

    if (didLoginSucceed) setUser(username);

    return didLoginSucceed;
  };

  const logout = (loggingOut) => {
    setUser(null);
    loggingOut();
  };

  return (
    <AuthContext.Provider value={{ user, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

const useAuth = () => {
  const context = useContext(AuthContext);

  if (context === undefined)
    throw new Error(
      'useAuth needs to be used on a descendant element of the Provider'
    );

  return context;
};

export { AuthProvider, useAuth };
