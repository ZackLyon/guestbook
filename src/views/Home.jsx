import React from 'react';
import CommentList from '../components/CommentList';
import Inputs from '../components/Inputs';

export default function Home() {
  return (
    <div>
      <Inputs />
      <CommentList />
    </div>
  );
}
