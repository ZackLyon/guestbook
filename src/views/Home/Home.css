section {
  min-height: 90vh;
  display: grid;
  grid-template-rows: 25vh 4fr;
  background-color: black;
}
