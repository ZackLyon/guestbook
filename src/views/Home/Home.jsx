import React from 'react';
import CommentList from '../../components/CommentList/CommentList';
import Inputs from '../../components/Inputs/Inputs';
import { NoteProvider } from '../../context/NoteContext';
import './Home.css';

export default function Home() {
  return (
    <div>
      <NoteProvider>
        <section className="main-section">
          <Inputs />
          <CommentList />
        </section>
      </NoteProvider>
    </div>
  );
}
