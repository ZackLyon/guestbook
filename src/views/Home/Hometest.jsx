import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from './Home';
import { useUser } from '../../context/UserContext.jsx';

it('should display Home view on the screen', async () => {
  const { container } = await render(<Home />);

  screen.debug();

  expect(container).toMatchSnapshot();
});
