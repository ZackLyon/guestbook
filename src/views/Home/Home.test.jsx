import { render } from '@testing-library/react';
import Home from './Home';
import { AuthProvider } from '../../context/AuthContext.jsx';

it('should display Home view on the screen', () => {
  const { container } = render(
    <AuthProvider>
      <Home />
    </AuthProvider>
  );

  expect(container).toMatchSnapshot();
});
