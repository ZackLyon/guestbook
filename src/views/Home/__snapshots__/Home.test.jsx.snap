// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display Home view on the screen 1`] = `
<div>
  <div>
    <section
      class="main-section"
    >
      <div
        class="inputContainer"
      >
        <div
          class="messageControls"
        >
          <form
            class="messageForm"
          >
            <input
              placeholder="message"
              type="textarea"
              value=""
            />
            <button
              type="submit"
            >
              Submit
            </button>
          </form>
          <button>
            Sign Out
          </button>
        </div>
      </div>
      <div
        class="commentContainer"
      />
    </section>
  </div>
</div>
`;
